tests/test_simple.py

```diff
@@ -103,6 +115,38 @@ def test_workers_health_custom_path(use_zmq):
         assert response.text == "ok"
 
 
+@pytest.mark.parametrize("use_zmq", [True, False])
+def test_workers_health_with_custom_health_method(use_zmq):
+    server = LitServer(
+        SlowSetupWithCustomHealthLitAPI(),
+        accelerator="cpu",
+        devices=1,
+        timeout=5,
+        workers_per_device=2,
+        fast_queue=use_zmq,
+    )
+
+    with wrap_litserve_start(server) as server, TestClient(server.app) as client:
+        response = client.get("/health")
+        assert response.status_code == 503
+        assert response.text == "not ready"
+
+        time.sleep(1)
+        response = client.get("/health")
+        assert response.status_code == 503
+        assert response.text == "not ready"
+
+        time.sleep(3)
```

Comment:
let's only sleep 1 here and sleep 4 below. Depending on the time the test and client takes for setup, this might become flaky otherwise.

Reply:
Sure